### PR TITLE
refactor: increase readability and add tests

### PR DESCRIPTION
**Pull request by aleortega**
This PR refactors the service to increase readability. Also, it adds validations and error messages to ensure robustness.
Finally, increase code coverage by adding tests, more are coming in further pull requests.

Pendings:
- [ ] Add new property relating original size with new size
- [ ] Write document detailing the auth options
- [ ] Implement auth workflow
- [ ] Implement cache to also cache error conversions
- [ ] Metrics

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -10,41 +10,40 @@ RUN apt-get -y -qq install make build-essential git
...
 
 WORKDIR /app
 
-# some packages require a build step
+# Install environment dependencies
 RUN apt-get update
 RUN apt-get -y -qq install python-setuptools build-essential gnupg curl
 
-# We use Tini to handle signals and PID1 (https://github.com/krallin/tini, read why here https://github.com/krallin/tini/issues/8)
+## We use Tini to handle signals and PID1 (https://github.com/krallin/tini, read why here https://github.com/krallin/tini/issues/8)
```
Comment (marianogoldman) [inline Dockerfile:22]: There are few places were there's an extra `#` added.

File: src/adapters/cdnBucket.ts [hunk 2]
```diff
@@ -0,0 +1,27 @@
+import * as AWS from 'aws-sdk'
+import * as MockAws from 'mock-aws-s3'
+import * as stream from 'stream'
+import { AppComponents } from '../types'
+import ICDNBucket from '../ports/ICDNBucket'
+
+export async function createCDNBucket({ config }: Pick<AppComponents, 'config'>): Promise<ICDNBucket> {
+  const bucketName = await config.getString('BUCKET')
+  const bucket = bucketName ? new AWS.S3({}) : new MockAws.S3({})
```
[the remark this answers is not in the document]
Comment (marianogoldman) [reply src/adapters/cdnBucket.ts:9]: This approach allows to configure the mock when the app is running in prod...

File: src/adapters/commandLine.ts [hunk 3]
```diff
@@ -1,10 +1,24 @@
...
-import ProcessWrapper from '../types/processWrapper'
+import ICommandLine from '../ports/ICommandTrigger'
+import { AppComponents } from './../types'
+import { ConversionResult } from '../types/asset-converter'
 
-export default function createCommandTrigger(): ICommandTrigger {
-  function execute(command: string, args: string[]): ProcessWrapper {
-    return new ProcessWrapper(spawn(command, args))
+export default async function createCommandLineAdapter({ logs }: Pick<AppComponents, 'logs'>): Promise<ICommandLine> {
+  const logger = logs.getLogger('commandLineAdapter')
```
Comment (marianogoldman) [inline src/adapters/commandLine.ts:7]: Let's unify all logger names either on kebab-case names or camel-case (we're using kebab in another one further below). I personally prefer kebab, but up to you as long they're consistent.

**Review by marianogoldman — CHANGES_REQUESTED** [review]

File: Dockerfile [hunk 4]
```diff
@@ -10,41 +10,40 @@ RUN apt-get -y -qq install make build-essential git
...
 RUN yarn install --frozen-lockfile
 
-# build the app
+## Build app
 COPY . /app
 RUN yarn run build
 RUN yarn run test
 
-# remove devDependencies, keep only used dependencies
+## Tree-shake dev dependencies
```
[the remark this answers is not in the document]
Comment (aleortega) [reply Dockerfile:43]: Oh I thought this line was removing the devs dependencies installed on line 36, thanks!

File: test/unit/fileSystem.spec.ts [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+    })
+    const logs = createLogComponentMock()
+
+    sut = await createFileSystemAdapter({ config, logs })
+  })
+
+  describe('saveFile should', () => {
+    const writeFileMock = jest.spyOn(fs.promises, 'writeFile').mockResolvedValue()
+
+    afterEach(() => {
```
[the remark this answers is not in the document]
Comment (aleortega) [reply test/unit/fileSystem.spec.ts:26]: I'm resetting different mocks on each describe, that's why they are at this level. Each describes only spies on the function it is consuming

File: Dockerfile (hunk 4 above)
[the remark this answers is not in the document]
Comment (marianogoldman) [reply Dockerfile]: It does. But that's not tree-shaking...

**Review by marianogoldman — APPROVED** [review]

**Pull request merged**